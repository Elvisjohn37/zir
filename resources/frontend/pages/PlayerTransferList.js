import React from 'react';
import styles from './PlayerTransferList.module';
import { LangDataProvider } from 'frontend/components/Language';
import { FormattedMessage } from 'react-intl';
import { Date, DateTime } from 'frontend/components/ReportCommonFormatter';
import TableContainer from '@mui/material/TableContainer';
import Table from '@mui/material/Table';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import Sidebar from 'frontend/components/Sidebar';
import TableIndicator from 'frontend/components/TableIndicator';
import { Description, Amount, OutstandingBalance, PlayableBalance } from 'frontend/components/ReportTransferFormatter';
import { UserConsumer } from 'frontend/contexts/User';
import { MobileTableRefresh, DesktopTableRefresh } from 'frontend/components/TableRefresh';
import MediaQuery from 'react-responsive';
import usePlayerTransferList from './playertransferlist/usePlayerTransferList';
import { LayoutConsumer } from 'frontend/contexts/Layout';
import Spinner from 'frontend/components/Spinner';

export default function PlayerTransferList(props) {
	let params = props.match.params;

	let [isFetching, setIsFetching] = React.useState(true);
	let [activePage, setActivePage] = React.useState(0);

	let [transferListError, setTransferListError] = React.useState(false);
	let playerTransferList = usePlayerTransferList({
		isFetching,
		setIsFetching,
		date: params.date,
		page: activePage + 1,
		setTransferListError,
	});

	return (
		<LayoutConsumer>
			{({ layoutState }) => (
				<LangDataProvider category="reports" fallback={<Spinner/>}>
					<Sidebar sidebarGroup="archivedReport" disabled={layoutState.config.isMobileSite}>
						<div className={styles.tableSection}>
							<MediaQuery maxDeviceWidth={styles.main_mobile} className={styles.mediaQueryButton}>
								{(matches) =>
									matches ? (
										<MobileTableRefresh
											onClick={() => setIsFetching(true)}
											isDisabled={isFetching}
											display={<Date date={params.date} />}
										/>
									) : (
										<DesktopTableRefresh
											onClick={() => setIsFetching(true)}
											isDisabled={isFetching}
											id="creditDetailRefresh"
											values={{ date: <Date date={params.date} /> }}
										/>
									)
								}
							</MediaQuery>

							<div>
								<TableIndicator
									id="testTableInit"
									isLoading={isFetching}
									isEmpty={playerTransferList.content.length == 0}
									isError={transferListError}
									pagination={{
										count: playerTransferList.totalRow,
										page: activePage,
										rowsPerPage: playerTransferList.rowsPerPage,
										onPageChange: (event, page) => {
											setActivePage(page);
											setIsFetching(true);
										},
									}}
								>
									<TableContainer>
										<Table classes={{ root: styles.table }}>
											<TableHead>
												<TableRow classes={{ root: styles.row }}>
													<TableCell align="center" className={styles.colHeaderRowNum}>
														#
													</TableCell>
													<TableCell align="center" className={styles.colHeaderDateTime}>
														<FormattedMessage id="date" /> &amp; <FormattedMessage id="time" />
													</TableCell>
													<TableCell align="center" className={styles.colHeaderDescription}>
														<FormattedMessage id="details" />
													</TableCell>
													<TableCell align="center" className={styles.colHeaderAmount}>
														<FormattedMessage id="amount" />
													</TableCell>
													<TableCell align="center" className={styles.colHeaderOutstandingBalance}>
														<FormattedMessage id="outstandingBalance" />
													</TableCell>
													<TableCell align="center" className={styles.colHeaderPlayableBalance}>
														<FormattedMessage id="playableBalance" />
													</TableCell>
												</TableRow>
											</TableHead>
											<UserConsumer>
												{({ userState }) => {
													return (
														<TableBody id="testTableBody">
															{playerTransferList.content.map((row) => (
																<TableRow key={row.rowNo} classes={{ root: styles.row }} className="testTableRow">
																	<TableCell className="testTableCell" align="center" scope="row">
																		{row.rowNo}
																	</TableCell>
																	<TableCell className="testTableCell" align="center">
																		<DateTime dateTime={row.dateTime} newLine={true} className="testTableDateTime" />
																	</TableCell>
																	<TableCell className="testTableCell" align="center">
																		<Description
																			className="testDescription"
																			row={row}
																			agentUsername={userState.user.parent}
																		/>
																	</TableCell>
																	<TableCell className="testTableCell" align="right">
																		<Amount row={row} className="testAmount" />
																	</TableCell>
																	<TableCell className="testTableCell" align="right">
																		<OutstandingBalance row={row} className="testOutstandingBalance" />
																	</TableCell>
																	<TableCell className="testTableCell" align="right">
																		<PlayableBalance row={row} className="testPlayableBalance" />
																	</TableCell>
																</TableRow>
															))}
														</TableBody>
													);
												}}
											</UserConsumer>
										</Table>
									</TableContainer>
								</TableIndicator>
							</div>
						</div>
					</Sidebar>
				</LangDataProvider>
			)}
		</LayoutConsumer>
	);
}
