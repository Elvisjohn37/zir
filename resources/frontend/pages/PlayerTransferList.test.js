import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import PlayerTransferList from './PlayerTransferList';
import * as MediaQuery from 'react-responsive';
import { setUserState } from 'frontend/contexts/User';
import * as usePlayerTransferList from './playertransferlist/usePlayerTransferList';
import userEvent from '@testing-library/user-event';

jest.mock('./playertransferlist/usePlayerTransferList');
jest.mock('./playertransferlist/usePlayerTransferList');
jest.mock('@mui/material/Table');
jest.mock('@mui/material/TableHead');
jest.mock('@mui/material/TableRow');
jest.mock('@mui/material/TableBody');
jest.mock('@mui/material/TableCell');
jest.mock('@mui/material/TableContainer');
jest.mock('frontend/components/TableIndicator');
jest.mock('frontend/components/Sidebar');
jest.mock('frontend/components/Language');
jest.mock('frontend/components/ReportCommonFormatter');
jest.mock('frontend/components/ReportTransferFormatter');
jest.mock('frontend/components/TableRefresh');
jest.mock('frontend/contexts/Layout');
jest.mock('frontend/contexts/Language');
jest.mock('frontend/contexts/User');
jest.mock('react-responsive');
jest.mock('react-intl');

describe('<PlayerTransferList />', () => {
	let defaultProps = {
		match: {
			params: {
				date: '2021-07-15',
			},
		},
	};

	afterEach(() => {
		resetMocks();
		cleanup();
	});

	function resetMocks() {
		jest.clearAllMocks();
		jest.restoreAllMocks();
		usePlayerTransferList.resetUsePlayerTransferListData();
		MediaQuery.setMediaQuery(true);
	}

	function expectCorrectTableContent() {
		setUserState({ user: { parent: 'test' } });
		let responseData = usePlayerTransferList.getPlayerTransferListData().data.data;
		let columnsLength = 6;
		responseData.content.forEach((data, index) => {
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 1]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 2]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 3]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 4]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 5]).toBeInTheDocument();
		});
	}

	it('should match snapshot', () => {
		let { asFragment } = render(<PlayerTransferList {...defaultProps} />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should have table loading indicator on table during data fetching', () => {
		usePlayerTransferList.setIsProceedHook(false);
		render(<PlayerTransferList {...defaultProps} />);
		expect(screen.getByText('Loading TableIndicator')).toBeInTheDocument();
	});

	it('should display correct data in table after request from backend', () => {
		let { rerender } = render(<PlayerTransferList {...defaultProps} />);
		expectCorrectTableContent();
		usePlayerTransferList.setUsePlayerTransferListResponse(true, {
			content: [
				{
					rowNo: '7',
					amount: '1000.000000',
					amountFormat: 'defaultAmount',
					dateTime: '2021-07-15 14:32:23.753000',
					descriptionFormat: 'depositApproved',
					outstandingBalanceFormat: 'noOutstandingBalance',
					playableBalance: '1000.000000',
				},
				{
					rowNo: '8',
					amount: '1000.000000',
					amountFormat: 'defaultAmount',
					dateTime: '2021-07-15 14:32:23.753000',
					descriptionFormat: 'depositApproved',
					outstandingBalanceFormat: 'noOutstandingBalance',
					playableBalance: '1000.000000',
				},
			],
			totalRow: 2,
			rowsPerPage: 20,
		});
		rerender(<PlayerTransferList {...defaultProps} />);
		expectCorrectTableContent();
	});

	it('should fetch data from backend when user click the refresh button', () => {
		render(<PlayerTransferList {...defaultProps} />);
		userEvent.click(screen.getByTestId('MobileTableRefresh'));
		expect(usePlayerTransferList.default).toHaveBeenNthCalledWith(1, {
			isFetching: true,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			setTransferListError: expect.any(Function),
		});
		expect(usePlayerTransferList.default).toHaveBeenNthCalledWith(2, {
			isFetching: false,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			setTransferListError: expect.any(Function),
		});
		expect(usePlayerTransferList.default).toHaveBeenNthCalledWith(3, {
			isFetching: true,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			setTransferListError: expect.any(Function),
		});
		expect(usePlayerTransferList.default).toHaveBeenNthCalledWith(4, {
			isFetching: false,
			setIsFetching: expect.any(Function),
			date: '2021-07-15',
			page: 1,
			setTransferListError: expect.any(Function),
		});
		expect(usePlayerTransferList.default).toHaveBeenCalledTimes(4);
	});

	it('should display refresh button base on the type of device', () => {
		let { rerender } = render(<PlayerTransferList {...defaultProps} />);
		let MobileTableRefresh = screen.getByTestId('MobileTableRefresh');
		expect(MobileTableRefresh).toBeInTheDocument();

		MediaQuery.setMediaQuery(false);
		rerender(<PlayerTransferList {...defaultProps} />);
		let DesktopTableRefresh = screen.getByTestId('DesktopTableRefresh');
		expect(DesktopTableRefresh).toBeInTheDocument();
	});

	it('should display pagination and record base on selected page', () => {
		render(<PlayerTransferList {...defaultProps} />);
		expect(screen.getByText('Pagination page: 0')).toBeInTheDocument();
		expect(
			screen.getByText('Pagination count: ' + usePlayerTransferList.usePlayerTransferListData.totalRow)
		).toBeInTheDocument();
		expect(
			screen.getByText('Pagination rowsPerPage: ' + usePlayerTransferList.usePlayerTransferListData.rowsPerPage)
		).toBeInTheDocument();
	});

	it('should display no record found when record is empty', () => {
		usePlayerTransferList.setUsePlayerTransferListResponse(true, { content: [] });
		let { container } = render(<PlayerTransferList {...defaultProps} />);
		let tableRow = container.getElementsByClassName('testTableRow');
		expect(tableRow).toHaveLength(0);
	});

	it('should display error message when backend response is not valid', () => {
		usePlayerTransferList.setUsePlayerTransferListResponse(false);
		render(<PlayerTransferList {...defaultProps} />);
		expect(screen.getByText('Error Response')).toBeInTheDocument();
	});
});
