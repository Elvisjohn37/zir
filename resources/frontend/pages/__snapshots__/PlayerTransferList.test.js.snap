// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerTransferList /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    category="reports"
    data-testid="LangDataProvider"
  >
    <div>
      reports
    </div>
    <div
      data-testid="default"
      sidebargroup="report"
    >
      <div
        class="tableSection"
      >
        <div
          data-testid="MobileTableRefresh"
          isdisabled="false"
        />
        <div>
          <div
            data-testid="TableIndicator"
            id="testTableInit"
            isempty="false"
            iserror="false"
            isloading="false"
            pagination="{\\"count\\":6,\\"page\\":0,\\"rowsPerPage\\":20}"
          >
            <span>
              Pagination count: 6
            </span>
            <span>
              Pagination page: 0
            </span>
            <span>
              Pagination rowsPerPage: 20
            </span>
            <span>
              Done loading TableIndicator
            </span>
            <div
              data-testid="TableContainer"
            >
              <div
                aria-label="a dense table"
                classes="{\\"root\\":\\"table\\"}"
                data-testid="Table"
              >
                <div
                  data-testid="TableHead"
                >
                  <div
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="colHeaderRowNum"
                      data-testid="TableCell"
                    >
                      #
                    </div>
                    <div
                      align="center"
                      class="colHeaderDateTime"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="date"
                      >
                        <div>
                          date
                        </div>
                      </div>
                       & 
                      <div
                        data-testid="FormattedMessage"
                        id="time"
                      >
                        <div>
                          time
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colHeaderDescription"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="details"
                      >
                        <div>
                          details
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colHeaderAmount"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="amount"
                      >
                        <div>
                          amount
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colHeaderOutstandingBalance"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="outstandingBalance"
                      >
                        <div>
                          outstandingBalance
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colHeaderPlayableBalance"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="playableBalance"
                      >
                        <div>
                          playableBalance
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  data-testid="TableBody"
                  id="testTableBody"
                >
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      11
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                        row="{\\"rowNo\\":\\"11\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        <div>
                          depositApproved1
                        </div>
                        <div>
                          testParent
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testAmount"
                        data-testid="Amount"
                        row="{\\"rowNo\\":\\"11\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        No Amount Found
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testOutstandingBalance"
                        data-testid="OutstandingBalance"
                        row="{\\"rowNo\\":\\"11\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        No Outstanding Balance Found
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                        row="{\\"rowNo\\":\\"11\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      21
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                        row="{\\"rowNo\\":\\"21\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        <div>
                          depositApproved1
                        </div>
                        <div>
                          testParent
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testAmount"
                        data-testid="Amount"
                        row="{\\"rowNo\\":\\"21\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        No Amount Found
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testOutstandingBalance"
                        data-testid="OutstandingBalance"
                        row="{\\"rowNo\\":\\"21\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        No Outstanding Balance Found
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                        row="{\\"rowNo\\":\\"21\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      31
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                        row="{\\"rowNo\\":\\"31\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        <div>
                          depositApproved1
                        </div>
                        <div>
                          testParent
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testAmount"
                        data-testid="Amount"
                        row="{\\"rowNo\\":\\"31\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        No Amount Found
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testOutstandingBalance"
                        data-testid="OutstandingBalance"
                        row="{\\"rowNo\\":\\"31\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        No Outstanding Balance Found
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                        row="{\\"rowNo\\":\\"31\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      41
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                        row="{\\"rowNo\\":\\"41\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        <div>
                          depositApproved1
                        </div>
                        <div>
                          testParent
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testAmount"
                        data-testid="Amount"
                        row="{\\"rowNo\\":\\"41\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        No Amount Found
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testOutstandingBalance"
                        data-testid="OutstandingBalance"
                        row="{\\"rowNo\\":\\"41\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        No Outstanding Balance Found
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                        row="{\\"rowNo\\":\\"41\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      51
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                        row="{\\"rowNo\\":\\"51\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        <div>
                          depositApproved1
                        </div>
                        <div>
                          testParent
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testAmount"
                        data-testid="Amount"
                        row="{\\"rowNo\\":\\"51\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        No Amount Found
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testOutstandingBalance"
                        data-testid="OutstandingBalance"
                        row="{\\"rowNo\\":\\"51\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        No Outstanding Balance Found
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                        row="{\\"rowNo\\":\\"51\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                  <div
                    class="testTableRow"
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                      scope="row"
                    >
                      61
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        agentusername="testParent"
                        class="testDescription"
                        data-testid="Description"
                        row="{\\"rowNo\\":\\"61\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        <div>
                          depositApproved1
                        </div>
                        <div>
                          testParent
                        </div>
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testAmount"
                        data-testid="Amount"
                        row="{\\"rowNo\\":\\"61\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        No Amount Found
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testOutstandingBalance"
                        data-testid="OutstandingBalance"
                        row="{\\"rowNo\\":\\"61\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        No Outstanding Balance Found
                      </div>
                    </div>
                    <div
                      align="right"
                      class="testTableCell"
                      data-testid="TableCell"
                    >
                      <div
                        class="testPlayableBalance"
                        data-testid="PlayableBalance"
                        row="{\\"rowNo\\":\\"61\\",\\"amount\\":\\"1000.0000001\\",\\"amountFormat\\":\\"defaultAmount1\\",\\"dateTime\\":\\"2021-07-15 14:32:23.7530001\\",\\"descriptionFormat\\":\\"depositApproved1\\",\\"outstandingBalanceFormat\\":\\"noOutstandingBalance1\\",\\"playableBalance\\":\\"1000.0000001\\"}"
                      >
                        1000.0000001
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
